torch/_inductor/graph.py

```diff
@@ -385,7 +385,7 @@ def run_node(self, n: torch.fx.Node):
             # Realize if (1) any user need inputs realized, or (2) there is
             # already too many reads and rematerializing can be bad.
             num_users = len(set(n.users))
-            if num_users > 1 and isinstance(result, TensorBox):
+            if num_users > 0 and isinstance(result, TensorBox):
```

Comment:
I'm a little wary of changing recomputation heuristics in this PR. Could you change in a separate pr please ?